# Big Data Assignment — Reading Files, Cleaning, and Data Management
### Course: Algorithm Analysis (Gpo 104)
### Student: Kira Dayanna Espinoza Ortiz - A00841144
### Date: August 24, 2026

In [1]:
import pandas as pd
import numpy as np
## 1. Loading the Dataset
mis_datos = pd.read_csv("../data/cars.csv")
## Question 1. Review the structure


In [2]:
mis_datos.shape

(428, 15)

As we can see, the sahpe of my data set is 428 rows and 15 columns.

In [4]:
mis_datos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 428 entries, 0 to 427
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Make         428 non-null    object 
 1   Model        428 non-null    object 
 2   Type         428 non-null    object 
 3   Origin       428 non-null    object 
 4   DriveTrain   428 non-null    object 
 5   MSRP         428 non-null    int64  
 6   Invoice      428 non-null    int64  
 7   EngineSize   425 non-null    float64
 8   Cylinders    425 non-null    float64
 9   Horsepower   423 non-null    float64
 10  MPG_City     426 non-null    float64
 11  MPG_Highway  426 non-null    float64
 12  Weight       423 non-null    float64
 13  Wheelbase    426 non-null    float64
 14  Length       427 non-null    float64
dtypes: float64(8), int64(2), object(5)
memory usage: 50.3+ KB


From this quick info check we can observe that some rows have less registered values than 428, which means we have missing values. We can also see the type of data registered in each row, in this data set, we have text objects, integers (whole numbers), and float numbers (with decimals).

In [4]:
mis_datos.columns

Index(['Make', 'Model', 'Type', 'Origin', 'DriveTrain', 'MSRP', 'Invoice',
       'EngineSize', 'Cylinders', 'Horsepower', 'MPG_City', 'MPG_Highway',
       'Weight', 'Wheelbase', 'Length'],
      dtype='object')

We can see the names of the 15 columns

In [5]:
mis_datos.head()

,Make,Model,Type,Origin,DriveTrain,MSRP,Invoice,EngineSize,Cylinders,Horsepower,MPG_City,MPG_Highway,Weight,Wheelbase,Length
0,Acura,MDX,SUV,Asia,All,36945,33337,3.5,6.0,265.0,17.0,23.0,4451.0,106.0,189.0
1,Acura,RSX Type S 2dr,Sedan,Asia,Front,23820,21761,2.0,4.0,200.0,24.0,31.0,2778.0,101.0,172.0
2,Acura,TSX 4dr,Sedan,Asia,Front,26990,24647,2.4,4.0,200.0,22.0,29.0,3230.0,105.0,183.0
3,Acura,TL 4dr,Sedan,Asia,Front,33195,30299,3.2,6.0,270.0,20.0,28.0,NaN,108.0,186.0
4,Acura,3.5 RL 4dr,Sedan,Asia,Front,43755,39014,3.5,6.0,225.0,18.0,24.0,3880.0,115.0,197.0


Here we can observe the first rows of data

In [6]:
mis_datos.tail()

,Make,Model,Type,Origin,DriveTrain,MSRP,Invoice,EngineSize,Cylinders,Horsepower,MPG_City,MPG_Highway,Weight,Wheelbase,Length
423,Volvo,C70 LPT convertible 2dr,Sedan,Europe,Front,40565,38203,2.4,5.0,197.0,21.0,28.0,3450.0,105.0,186.0
424,Volvo,C70 HPT convertible 2dr,Sedan,Europe,Front,42565,40083,2.3,5.0,242.0,20.0,26.0,3450.0,105.0,186.0
425,Volvo,S80 T6 4dr,Sedan,Europe,Front,45210,42573,2.9,6.0,268.0,19.0,26.0,3653.0,110.0,190.0
426,Volvo,V40,Wagon,Europe,Front,26135,24641,1.9,4.0,170.0,22.0,29.0,2822.0,101.0,180.0
427,Volvo,XC70,Wagon,Europe,All,35145,33112,2.5,5.0,208.0,20.0,27.0,3823.0,109.0,186.0


And here the last rows of data.

In [5]:
mis_datos.sample(5)

,Make,Model,Type,Origin,DriveTrain,MSRP,Invoice,EngineSize,Cylinders,Horsepower,MPG_City,MPG_Highway,Weight,Wheelbase,Length
133,Ford,Mustang 2dr (convertible),Sports,USA,Rear,18345,16943,3.8,6.0,193.0,20.0,29.0,3290.0,101.0,183.0
154,Honda,Civic DX 2dr,Sedan,Asia,Front,13270,12175,1.7,4.0,115.0,32.0,38.0,2432.0,103.0,175.0
256,Mercedes-Benz,C240 4dr,Sedan,Europe,All,33480,31187,2.6,6.0,168.0,19.0,25.0,3360.0,107.0,178.0
150,Honda,Insight 2dr (gas/electric),Hybrid,Asia,Front,19110,17911,2.0,3.0,73.0,60.0,66.0,1850.0,95.0,155.0
292,Mitsubishi,Lancer OZ Rally 4dr auto,Sedan,Asia,Front,17232,16196,2.0,4.0,120.0,25.0,31.0,2744.0,102.0,181.0


We can also get a random sample of 5 rows from the dataset. 

In [9]:
mis_datos.describe()

,MSRP,Invoice,EngineSize,Cylinders,Horsepower,MPG_City,MPG_Highway,Weight,Wheelbase,Length
count,428.000000,428.000000,425.000000,425.000000,423.000000,426.000000,426.000000,423.000000,426.000000,427.000000
mean,32774.855140,30014.700935,3.195294,5.811765,215.576832,20.065728,26.840376,3578.229314,108.166667,186.381733
std,19431.716674,17642.117750,1.111932,1.557802,71.978098,5.249574,5.749505,759.034549,8.327582,14.369110
min,10280.000000,9875.000000,1.300000,3.000000,73.000000,10.000000,12.000000,1850.000000,89.000000,143.000000
25%,20334.250000,18866.000000,2.300000,4.000000,165.000000,17.000000,24.000000,3103.000000,103.000000,178.000000
50%,27635.000000,25294.500000,3.000000,6.000000,210.000000,19.000000,26.000000,3476.000000,107.000000,187.000000
75%,39205.000000,35710.250000,3.900000,6.000000,252.500000,21.750000,29.000000,3978.500000,112.000000,194.000000
max,192465.000000,173560.000000,8.300000,12.000000,500.000000,60.000000,66.000000,7190.000000,144.000000,238.000000


We can get a quick statistical summary of the data.

## Question 2 Check for missing values and replace them with the mean.

In this part, we first check how many missing values we have in each column.

In [11]:
mis_datos.isna().sum()

Make           0
Model          0
Type           0
Origin         0
DriveTrain     0
MSRP           0
Invoice        0
EngineSize     3
Cylinders      3
Horsepower     5
MPG_City       2
MPG_Highway    2
Weight         5
Wheelbase      2
Length         1
dtype: int64

As we can see, we have several missing values, in EngineSize, Cylinders, Horsepower, MPG_City       2
MPG_Highway    2
Weight         5
Wheelbase      2
Length         1

In [12]:
# EngineSize
average_enginesize = mis_datos["EngineSize"].mean()
mis_datos["EngineSize"] = (
    mis_datos["EngineSize"]
    .fillna(average_enginesize)
)

# Cylinders
average_cylinders = mis_datos["Cylinders"].mean()
mis_datos["Cylinders"] = (
    mis_datos["Cylinders"]
    .fillna(average_cylinders)
)

# Horsepower
average_horsepower = mis_datos["Horsepower"].mean()
mis_datos["Horsepower"] = (
    mis_datos["Horsepower"]
    .fillna(average_horsepower)
)

# MPG_City
average_mpg_city = mis_datos["MPG_City"].mean()
mis_datos["MPG_City"] = (
    mis_datos["MPG_City"]
    .fillna(average_mpg_city)
)

# MPG_Highway
average_mpg_highway = mis_datos["MPG_Highway"].mean()
mis_datos["MPG_Highway"] = (
    mis_datos["MPG_Highway"]
    .fillna(average_mpg_highway)
)

# Weight
average_weight = mis_datos["Weight"].mean()
mis_datos["Weight"] = (
    mis_datos["Weight"]
    .fillna(average_weight)
)

# Wheelbase
average_wheelbase = mis_datos["Wheelbase"].mean()
mis_datos["Wheelbase"] = (
    mis_datos["Wheelbase"]
    .fillna(average_wheelbase)
)

# Length
average_length = mis_datos["Length"].mean()
mis_datos["Length"] = (
    mis_datos["Length"]
    .fillna(average_length)
)

In [13]:
mis_datos.isna().sum()

Make           0
Model          0
Type           0
Origin         0
DriveTrain     0
MSRP           0
Invoice        0
EngineSize     0
Cylinders      0
Horsepower     0
MPG_City       0
MPG_Highway    0
Weight         0
Wheelbase      0
Length         0
dtype: int64

## Question 3 Which manufacturer produces the most vehicles, and which produces the fewest?